### ISSUE TITLE
Objects merging doesn't work

### ISSUE THREAD

**Issue by eric239** [chimpler/pyhocon#10]
Using simple example from HOCON readme:
```
data-center-generic = { cluster-size = 6 }
data-center-east = ${data-center-generic} { name = "east" }
```
(saved as `test.conf`)

```
In [13]: conf = ConfigFactory.parse_file('test.conf')
---------------------------------------------------------------------------
ParseException                            Traceback (most recent call last)
<ipython-input-13-d3bb51afbb00> in <module>()
----> 1 conf = ConfigFactory.parse_file('test.conf')

/usr/local/lib/python2.7/site-packages/pyhocon/__init__.pyc in parse_file(filename)
     27         with open(filename, 'r') as fd:
     28             content = fd.read()
---> 29             return ConfigFactory.parse_string(content, os.path.dirname(filename))
     30
     31     @staticmethod

/usr/local/lib/python2.7/site-packages/pyhocon/__init__.pyc in parse_string(content, basedir)
     55         :type return: Config
     56         """
---> 57         return ConfigParser().parse(content, basedir)
     58
     59

/usr/local/lib/python2.7/site-packages/pyhocon/__init__.pyc in parse(content, basedir)
    190             + (list_expr | dict_expr | inside_dict_expr) \
    191             + ZeroOrMore(comment | eol_comma)
--> 192         config = config_expr.parseString(content, parseAll=True)[0]
    193         ConfigParser._resolve_substitutions(config, substitutions)
    194         return config

/usr/local/lib/python2.7/site-packages/pyparsing.pyc in parseString(self, instring, parseAll)
   1123             else:
   1124                 # catch and re-raise exception from here, clears out pyparsing internal stack trace
-> 1125                 raise exc
   1126         else:
   1127             return tokens

ParseException: Expected end of text (at char 85), (line:2, col:43)
```

Inheritance is a such a nice feature in HOCON, hopefully this can be fixed.

**Comment by eric239** [comment]
The environment:
```
→ python --version
Python 2.7.9

→ pip show pyhocon
---
Metadata-Version: 1.1
Name: pyhocon
Version: 0.2.4
Summary: HOCON parser for Python
Home-page: http://github.com/chimpler/pyhocon/
Author: Francois Dang Ngoc
Author-email: francois.dangngoc@gmail.com
License: Apache License 2.0
Location: /usr/local/lib/python2.7/site-packages
Requires: pyparsing
Entry-points:
  [console_scripts]
  pyhocon = pyhocon.tool:main

 |2.0.0-p451| ernestm-1762 in ~/workspace/pclick/conf
±  → pip show pyparsing
---
Metadata-Version: 2.0
Name: pyparsing
Version: 2.0.3
Summary: Python parsing module
Home-page: http://pyparsing.wikispaces.com/
Author: Paul McGuire
Author-email: ptmcg@users.sourceforge.net
License: MIT License
Location: /usr/local/lib/python2.7/site-packages
Requires:
```

**Comment by darthbear** [comment]
Thank you Ernest for the bug report. I'll work on it over the weekend!

**Comment by darthbear** [comment]
Ok, it's merged and pushed to pypi (pyhocon==0.2.5). Please let me know if you find any issue. Thanks @eric239 for the bug report again!

**Comment by eric239** [comment]
Awesome, thanks @darthbear for such a quick turn-around!
